auslib/test/admin/views/test_required_signoffs.py

```diff
@@ -0,0 +1,923 @@
+        data = {
+            "when": 400000000, "product": "foo", "channel": "bar", "role": "relman", "signoffs_required": 1, "change_type": "insert",
+        }
+        ret = self._post("/scheduled_changes/required_signoffs/product", data=data)
+        self.assertEquals(ret.status_code, 200, ret.data)
+        self.assertEquals(json.loads(ret.data), {"sc_id": 5})
+        r = dbo.productRequiredSignoffs.scheduled_changes.t.select().where(dbo.productRequiredSignoffs.scheduled_changes.sc_id == 5).execute().fetchall()
+        self.assertEquals(len(r), 1)
+        db_data = dict(r[0])
+        expected = {
+            "sc_id": 5, "scheduled_by": "bill", "change_type": "insert", "complete": False, "data_version": 1,
+            "base_product": "foo", "base_channel": "bar", "base_role": "relman", "base_signoffs_required": 1, "base_data_version": None,
+        }
+        self.assertEquals(db_data, expected)
+        cond = dbo.productRequiredSignoffs.scheduled_changes.conditions.t.select()\
+            .where(dbo.productRequiredSignoffs.scheduled_changes.conditions.sc_id == 5).execute().fetchall()
+        self.assertEquals(len(cond), 1)
+        cond_expected = {"sc_id": 5, "data_version": 1, "when": 400000000}
+        self.assertEquals(dict(cond[0]), cond_expected)
+
+    @mock.patch("time.time", mock.MagicMock(return_value=300))
+    def testAddScheduledChangeDeleteRequiredSignoff(self):
+        data = {
+            "when": 400000000, "product": "fake", "channel": "k", "role": "releng", "change_type": "delete", "data_version": 2,
+        }
+        ret = self._post("/scheduled_changes/required_signoffs/product", data=data)
+        self.assertEquals(ret.status_code, 200, ret.data)
+        self.assertEquals(json.loads(ret.data), {"sc_id": 5})
+        r = dbo.productRequiredSignoffs.scheduled_changes.t.select().where(dbo.productRequiredSignoffs.scheduled_changes.sc_id == 5).execute().fetchall()
+        self.assertEquals(len(r), 1)
+        db_data = dict(r[0])
+        expected = {
+            "sc_id": 5, "scheduled_by": "bill", "change_type": "delete", "complete": False, "data_version": 1,
+            "base_product": "fake", "base_channel": "k", "base_role": "releng", "base_signoffs_required": None, "base_data_version": 2,
+        }
+        self.assertEquals(db_data, expected)
+        cond = dbo.productRequiredSignoffs.scheduled_changes.conditions.t.select()\
+            .where(dbo.productRequiredSignoffs.scheduled_changes.conditions.sc_id == 5).execute().fetchall()
+        self.assertEquals(len(cond), 1)
+        cond_expected = {"sc_id": 5, "data_version": 1, "when": 400000000}
+        self.assertEquals(dict(cond[0]), cond_expected)
+
+    @mock.patch("time.time", mock.MagicMock(return_value=300))
+    def testUpdateScheduledChangeExistingRequiredSignoff(self):
+        data = {
+            "signoffs_required": 1, "data_version": 1, "sc_data_version": 1, "when": 200000000,
```

Comment:
Yes! This was implemented in https://bugzilla.mozilla.org/show_bug.cgi?id=1333095.